tests/common/util/NameGenerator.hpp

```diff
@@ -0,0 +1,28 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2024, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include <string>
+
+namespace tests::util {
+
+static auto const NameGenerator = [](auto const& info) { return info.param.testName; };
+
+}  // namespace tests::util
```

Comment:
Please set your IDE up to automatically add `\n` at end of files on save 👍

Reply:
That's just misinformation. Even your link says "no newline at end of file" in the accepted answer.
In my experience, the icon only ever appears if the newline is missing at the end of the file. I'm for consistency - the rest of the codebase (hopefully) follows the newline rule. And if we can enforce it with clang-format/tidy then we should do that as well 👍

Reply:
I created an issue #1434